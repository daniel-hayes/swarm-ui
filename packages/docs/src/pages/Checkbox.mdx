import { Checkbox } from 'swarm-components'
import Example from '../components/checkboxExample'

# Checkbox

```.jsx
<Checkbox label="baseball" checked={1 == 1} />
```

<Checkbox label="baseball" checked={1 == 1} />

<Example />