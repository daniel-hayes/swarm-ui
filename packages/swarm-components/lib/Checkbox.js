function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import * as React from 'react';
import Icon from './Icon';

// Can not inline css vars as color to Icon. Gray 5 icon fill on disbaled
const Checkbox = props => {
  const {
    checked,
    label,
    id = 'a',
    disabled,
    children,
    ...rest
  } = props;
  return React.createElement("label", _extends({
    "data-swarm-checkbox": disabled ? 'disabled' : 'default',
    for: id,
    disabled: disabled
  }, rest), React.createElement("span", {
    "data-swarm-checkbox-field": checked ? 'checked' : 'unchecked',
    tabIndex: "0",
    role: "checkbox",
    "aria-checked": checked
  }, checked && React.createElement(Icon, {
    shape: "check",
    size: "xs",
    color: disabled ? '#A2A2A2' : '#ffffff'
  })), React.createElement("input", {
    type: "checkbox",
    id: id,
    checked: checked,
    disabled: disabled
  }), React.createElement("span", null, label || children));
};

export default Checkbox;
Checkbox.__docgenInfo = {
  "description": "",
  "methods": [],
  "displayName": "Checkbox",
  "props": {
    "checked": {
      "required": true,
      "flowType": {
        "name": "boolean"
      },
      "description": ""
    },
    "disabled": {
      "required": false,
      "flowType": {
        "name": "boolean"
      },
      "description": ""
    },
    "id": {
      "required": false,
      "flowType": {
        "name": "string"
      },
      "description": ""
    },
    "name": {
      "required": true,
      "flowType": {
        "name": "string"
      },
      "description": ""
    }
  }
};