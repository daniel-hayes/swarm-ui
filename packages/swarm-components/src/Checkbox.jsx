import * as React from 'react';
import Icon from './Icon';

export type Props = {
	checked: boolean,
	disabled?: boolean,
	id?: string,
	name: string,
};

// Can not inline css vars as color to Icon. Gray 5 icon fill on disbaled
const Checkbox = (props: Props): React.ReactElement<'label'> => {
	const { checked, label, id = 'a', disabled, onChange, children, ...rest } = props;
	return (
		<label
			data-swarm-checkbox={disabled ? 'disabled' : 'default'}
			for={id}
			disabled={disabled}
			{...rest}
		>
			<span
				data-swarm-checkbox-field={checked ? 'checked' : 'unchecked'}
				role='checkbox'
				aria-checked={checked}
			>
				{checked && (
					<Icon shape="check" size="xs" color={disabled ? '#A2A2A2' : '#ffffff'} />
				)}
			</span>
			<input type="checkbox" id={id} checked={checked} disabled={disabled} onChange={onChange} />
			<span>{label || children}</span>
		</label>
	);
};

export default Checkbox;
