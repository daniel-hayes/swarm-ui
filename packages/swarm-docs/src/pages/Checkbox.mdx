import { Checkbox } from '@meetup/swarm-components'
import Example from '../components/checkboxExample'

# Checkbox

# Snippets
```.jsx
<>
<Checkbox label="swarm" checked={1 == 1} />
<Checkbox label="swarm" checked={1 == 1} />
</>
```

#Props


<Checkbox label="swarm" checked={1 == 1} />

<Example />