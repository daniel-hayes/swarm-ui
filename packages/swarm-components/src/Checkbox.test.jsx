import React from 'react';
import { shallow } from 'enzyme';
import Checkbox from './Checkbox';

describe('Checkbox', () => {
	const testCases = [
		['Default', <Checkbox key="1" >Check me</Checkbox>],
	];

	test.each(testCases)('Snapshot: %s', (description, element) => {
		expect(shallow(element)).toMatchSnapshot();
	});
});
