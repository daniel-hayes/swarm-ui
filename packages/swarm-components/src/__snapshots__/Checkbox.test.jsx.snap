// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox Snapshot: Checked 1`] = `
<label
  data-swarm-checkbox="default"
>
  <span
    aria-checked={true}
    data-swarm-checkbox-field="checked"
    role="checkbox"
  >
    <Icon
      color="#ffffff"
      shape="check"
      size="xs"
    />
  </span>
  <input
    checked={true}
    readOnly={true}
    type="checkbox"
  />
  <span>
    Uncheck me
  </span>
</label>
`;

exports[`Checkbox Snapshot: Default 1`] = `
<label
  data-swarm-checkbox="default"
>
  <span
    data-swarm-checkbox-field="unchecked"
    role="checkbox"
  />
  <input
    readOnly={true}
    type="checkbox"
  />
  <span>
    Check me
  </span>
</label>
`;

exports[`Checkbox Snapshot: Disbled checked 1`] = `
<label
  data-swarm-checkbox="disabled"
  disabled={true}
>
  <span
    aria-checked={true}
    data-swarm-checkbox-field="checked"
    role="checkbox"
  >
    <Icon
      color="#A2A2A2"
      shape="check"
      size="xs"
    />
  </span>
  <input
    checked={true}
    disabled={true}
    readOnly={true}
    type="checkbox"
  />
  <span>
    Can't uncheck me
  </span>
</label>
`;

exports[`Checkbox Snapshot: Disbled unchecked 1`] = `
<label
  data-swarm-checkbox="disabled"
  disabled={true}
>
  <span
    data-swarm-checkbox-field="unchecked"
    role="checkbox"
  />
  <input
    disabled={true}
    readOnly={true}
    type="checkbox"
  />
  <span>
    Can't check me
  </span>
</label>
`;
